### PR TITLE
feat: Asset buy page

### LINKED ISSUE #378: Add Item actions and "order" information to item detail
Add:
- Replace EXPIRES AT for `AVAILABLE: X`
- Mint number for wearables (need nft-server update). Only For L2
- Replace OWNER for AUTHOR/CREATOR
- Missing buy button
- Price

Not needed:
- BIDs

### LINKED ISSUE #379: Buy page for Items

### PR DESCRIPTION
**Pull request by nicosantangelo**
Closes #378
Closes #379

### CODE REVIEW

File: webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx [hunk 1]
```diff
@@ -15,8 +19,9 @@ import { WearableHighlights } from '../WearableHighlights'
 import { Props } from './ItemDetail.types'
 
 const ItemDetail = (props: Props) => {
-  const { item } = props
+  const { item, wallet } = props
   const wearable = item.data.wearable!
+  const canBuy = isOwnedBy(item, wallet) && !!item.price
```
Comment (cazala) [inline webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx:24]: Is this okay? it seems it's only letting you buy an item that is owned by you. I think we should rebase this branch and used the newly added `item.isOnSale` to determine if an item can be bought (there are items that are not for sale but still have a price, that's why `!!item.price` is not a reliable method)

File: webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx [hunk 2]
```diff
@@ -38,9 +43,34 @@ const ItemDetail = (props: Props) => {
         <Description text={wearable.description} />
         <Row>
           <Column align="left" grow={true}>
-            Order details here
+            {item.price ? (
```
Comment (cazala) [inline webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx:46]: Same as above, this should be the new `item.isOnSale` prop

File: webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx [hunk 3]
```diff
@@ -38,9 +43,34 @@ const ItemDetail = (props: Props) => {
...
+              <>
+                <Stats title={t('asset_page.price')}>
+                  <Mana network={item.network} withTooltip>
+                    {formatMANA(item.price)}
+                  </Mana>
+                </Stats>
+                <Stats title={t('asset_page.available')}>
+                  {item.available.toString()}
```
Comment (cazala) [inline webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx:54]: Maybe if the item is not on sale, but `available === 0` we should show something? I'm thinking in order to differentiate a "sold out" item from an item that's not on sale at all.

File: webapp/src/components/BuyPage/BuyItemModal/BuyItemModal.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  const handleClose = () => setShowAuthorizationModal(false)
+
+  const isDisabled = !item.price || isOwner || hasInsufficientMANA
+
+  const name = <Name asset={item} />
+
+  let subtitle = null
+  if (!item.price) {
```
Comment (cazala) [inline webapp/src/components/BuyPage/BuyItemModal/BuyItemModal.tsx:77]: same as above, `!item.isOnSale`

File: webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx (hunk 1 above)
Comment (nicosantangelo) [reply webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx:24]: Yes to both, I'll update the price check and negate the ownership

File: webapp/src/components/AssetPage/Owner/Owner.tsx [hunk 5]
```diff
@@ -9,21 +9,25 @@ import './Owner.css'
...
     vendor = asset.vendor
   } else {
+    label = t('asset_page.creator')
     address = asset.creator
     vendor = VendorName.DECENTRALAND
   }
 
   return (
     <div className="Owner">
       <Link to={locations.account(address, { vendor: vendor })}>
```
Comment (cazala) [inline webapp/src/components/AssetPage/Owner/Owner.tsx:29]: could we also add `resultType` and `section` here?  
- For an NFT `resultType=ResultType.NFT` and `section=Section[VendorName.DECENTRALAND].ALL`
- For an Item `resultType=ResultType.Item` and `section=Section[VendorName.DECENTRALAND].WEARABLE`

This way when you click on CREATOR in the detail you are taken to see all the creations by that creator

File: webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx (hunk 3 above)
Comment (nicosantangelo) [reply webapp/src/components/AssetPage/ItemDetail/ItemDetail.tsx:54]: I like the idea of a `sold out`! I'll add that very same thing and we can iterate later

**Review by cazala — APPROVED** [review]

**Pull request merged**